Q:

p2os connection

I have 3 pioneers, only one of which is running p2os. The two running Aria are all set up in terms of teleop and working with the navigation stack, I am having a problem with the Pioneer 2 with p2os.  I followed the tutorials carefully, but I have an error I can't seem to fix. I have an external laptop hooked up via USB2Serial connection.  I have a launch script like so
 <node name="p2os" pkg="p2os_driver" type="p2os" ns="$(env NAMESPACE)" output="screen" >
      <param name="port" value="/dev/Aria" />
      <remap from="/cmd_vel" to="/virtual_joystick/cmd_vel" />
      <remap from="pose" to="odom" />
   </node>

When I run the launch, it outputs the following message repeatedly for about a minute before crashing. I have tried changing permissions on my symlink, on ttyUSB*, and directly on the bus port its on, to no avail.
[ERROR] [1347049242.964984100][/robot_brain_3/p2os]: Error reading packet header from robot connection: P2OSPacket():Receive():read(): [Receive]

Also if you look on the display screen on the robot, it says
No Conn 9.6kb
p2os v1.p 14.8
Specifically the no conn presumably means no connection...but the output is different than if I just turned off the pioneer. It is clearly receiving something.
Oh and running the dashboard works, but it just spams the error in it's little logger, can't get any other useful information.
Any ideas?
Edit*
Sample process
sudo apt-get install ros-electric-p2os
sudo chmod 777 -R /dev/ttyUSB0
sudo chmod 777 -R /dev/Aria
sudo chmod 777 -R /dev/bus/usb/001/126  #these are all the same, just for sanity's sake
roslaunch p2os_launch sample_p2os.launch #where sample_p2os.launch is a script like above

[INFO] [/robot_brain_3/p2os]: using serial [port]: [/dev/Aria] [P2OSNODE]
[INFO] [/robot_brain_3/p2os]: P2OS connection opening serial port /dev/Aria...[Setup]
[ERROR] [/robot_brain_3/p2os]: Error reading packet header from robot connection: P2OSPacket():Receive():read(): [Receive]
[ERROR] [/robot_brain_3/p2os]: Error reading packet header from robot connection: P2OSPacket():Receive():read(): [Receive]
[ERROR] [/robot_brain_3/p2os]: Error reading packet header from robot connection: P2OSPacket():Receive():read(): [Receive]

Originally posted by Jyo on ROS Answers with karma: 100 on 2012-09-07
Post score: 3

Original comments
Comment by jbohren on 2012-09-09:
Does "/dev/Aria" exist, and what is it on this system if you're not using Aria?
Comment by Jyo on 2012-09-10:
`/dev/Aria does exist, and it is the usb2serial device to the pioneer. It is called Aria just for consistency sake, it could be renamed to /dev/pioneer or something.
Comment by jbohren on 2012-09-29:
Did you ever get this working?
Comment by jbohren on 2012-09-29:
What version of P2OS does your pioneer have? I found one with 1.D and it's giving the same error. Apparently the latest version is 1.Q
Comment by Jyo on 2012-10-01:
I have not gotten it working yet, it has been kind of sidelined, but I am using 1.P. I could try flashing the newest version, but it seems odd that that would be the issue :/
Comment by jbohren on 2012-10-01:
Yeah, if you're only one version off, it's probably not that. We spent some time hacking on a Pioneer2 over the weekend and got stuck with the same error you're reporting.
Comment by Mendelson on 2013-09-30:
Has anybody managed to solve this problem? I'm facing the same issue... I'm using a Pioneer P3-AT, Ubuntu 12.04 LTS, ROS Hydro.

A:

Add the user to the group dialout
Don't waist time to set rights on your usb devices.
On Ubuntu 12.04 it is enough to add the user to the group dialout

sudo usermod -a -G dialout $USER

The user have to logout and to login again to update the policy.
Greetings
Markus

Originally posted by Markus Bader with karma: 847 on 2012-10-10
This answer was ACCEPTED on the original site
Post score: 2

